shard-operator-parent/shard-operator-v2/src/main/java/com/redhat/service/smartevents/shard/operator/v2/controllers/ManagedProcessorController.java

```diff
@@ -0,0 +1,161 @@
+import com.redhat.service.smartevents.shard.operator.v2.resources.CamelIntegration;
+import com.redhat.service.smartevents.shard.operator.v2.resources.ManagedProcessor;
+import com.redhat.service.smartevents.shard.operator.v2.resources.ManagedProcessorStatus;
+
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.javaoperatorsdk.operator.api.reconciler.Context;
+import io.javaoperatorsdk.operator.api.reconciler.ControllerConfiguration;
+import io.javaoperatorsdk.operator.api.reconciler.DeleteControl;
+import io.javaoperatorsdk.operator.api.reconciler.ErrorStatusHandler;
+import io.javaoperatorsdk.operator.api.reconciler.EventSourceContext;
+import io.javaoperatorsdk.operator.api.reconciler.EventSourceInitializer;
+import io.javaoperatorsdk.operator.api.reconciler.Reconciler;
+import io.javaoperatorsdk.operator.api.reconciler.RetryInfo;
+import io.javaoperatorsdk.operator.api.reconciler.UpdateControl;
+import io.javaoperatorsdk.operator.processing.event.source.EventSource;
+
+@ApplicationScoped
+@ControllerConfiguration(labelSelector = LabelsBuilder.RECONCILER_LABEL_SELECTOR)
+public class ManagedProcessorController implements Reconciler<ManagedProcessor>,
+        EventSourceInitializer<ManagedProcessor>, ErrorStatusHandler<ManagedProcessor> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ManagedProcessorController.class);
+
+    @ConfigProperty(name = "event-bridge.executor.deployment.timeout-seconds")
+    int executorTimeoutSeconds;
+
+    @ConfigProperty(name = "event-bridge.executor.poll-interval.milliseconds")
+    int executorPollIntervalMilliseconds;
+
+    @Inject
+    KubernetesClient kubernetesClient;
+
+    @Inject
+    ManagedProcessorService managedProcessorService;
+
+    @Inject
+    BridgeErrorHelper bridgeErrorHelper;
+
+    @Inject
+    OperatorMetricsService metricsService;
+
+    @Override
+    public List<EventSource> prepareEventSources(EventSourceContext<ManagedProcessor> eventSourceContext) {
+
+        List<EventSource> eventSources = new ArrayList<>();
+        eventSources.add(EventSourceFactoryV2.buildIntegrationInformer(kubernetesClient, ManagedProcessor.COMPONENT_NAME));
+
+        return eventSources;
+    }
+
+    @Override
+    public UpdateControl<ManagedProcessor> reconcile(ManagedProcessor managedProcessor, Context context) {
+        LOGGER.info("Create or update ManagedProcessor: '{}' in namespace '{}'",
+                managedProcessor.getMetadata().getName(),
+                managedProcessor.getMetadata().getNamespace());
+
+        ManagedProcessorStatus status = managedProcessor.getStatus();
+
+        if (!status.isReady() && isTimedOut(status)) {
+            // notifyManagerOfFailure
+            status.markConditionFalse(ConditionTypeConstants.READY);
+            return UpdateControl.updateStatus(managedProcessor);
+        }
+
+        CamelIntegration camelIntegration = managedProcessorService.fetchOrCreateCamelIntegration(managedProcessor, "integrationName");
```

Comment:
@lucamolteni I assume that `integrationName` shouldn't be a hardcoded string? Any reason not to give it the same name as the `ManagedProcessor`